### PR TITLE
Add Heading to Admin

### PR DESCRIPTION
**Pull request by elanalynn**
### Background

Related to: https://github.com/Shopify/ui-extensions-private/issues/1464

This adds the `Heading` component for Admin UI Extensions, the implement in Web, and the example the UI Extensions component list. 

### Solution

This new `HeadingProps` API extends the `BaseHeadingProps` API and overrides the `level` property to support all HTML heading levels, similar to [Polaris](https://polaris.shopify.com/components/titles-and-text/heading). The default level is (h)2.

### 🎩

These instructions cover 3 related PRs:

- UI Extensions (this one)
- Web https://github.com/Shopify/web/pull/46630
- Admin UI Extensions https://github.com/Shopify/admin-ui-extensions/pull/116

#### Setup

- Check out the followign branches:
  - `1464/add-heading-to-admin-ui-extensions` in extensions-ui
  - `1464/heading` in admin-extensions-ui
  - `1464/heading-ui-extensions` in web
- In this branch (in `ui-extensions`), run the following:
  - `yarn build-consumer web`
  -  `yarn build-consumer admin-ui-extensions/packages/admin-ui-extensions-playground`
- In `admin-ui-extensions` run `dev up && dev s`
- In `web`, run `dev playground:argo` 
- Visit http://localhost:8080/#/playground/3p-components-list.worker.js

#### QA

Confirm that the `Heading` component appears in the example component list.
Screen Shot 2021-07-13 at 4 02 00 PM

 

### Checklist

- [x] I have :tophat:'d these changes

### CODE REVIEW

**Review by iainmcampbell — APPROVED** [review]

File: packages/admin-ui-extensions-react/src/components/Heading/examples/Heading.example.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import {extend, render, Heading} from '@shopify/admin-ui-extensions-react';
+
+function App() {
+  return <Heading>Heading</Heading>;
```
Comment (olavoasantos) [inline packages/admin-ui-extensions-react/src/components/Heading/examples/Heading.example.jsx:5]: Should the react/vanilla examples match?

Comment (elanalynn) [reply packages/admin-ui-extensions-react/src/components/Heading/examples/Heading.example.jsx]: Absolutely! Just a miss on my part. :)

File: packages/admin-ui-extensions-react/src/components/Heading/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export {Heading} from './Heading';
+export type {HeadingProps} from '@shopify/admin-ui-extensions';
```
[the remark this answers is not in the document]
Comment (elanalynn) [reply packages/admin-ui-extensions-react/src/components/Heading/index.ts:2]: From that ^ convo, it looks like @alanthai Will create a separate ticket to make all components using children consistent.

[the remark this answers is not in the document]
Comment (alanthai) [reply packages/admin-ui-extensions-react/src/components/Heading/index.ts:2]: ticket here https://github.com/orgs/Shopify/projects/1729#card-65737088

**Review by olavoasantos — APPROVED** [review]
There are some conflicts with `admin-rc` but the PR looks good! 🥇

**Pull request merged**